syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -297,3 +267,53 @@ func (j *archiveParser) discoverPkgsFromNestedArchives(parentPkg *pkg.Package) (
 
 	return pkgs, nil
 }
+
+func packageIdentitiesMatch(p pkg.Package, parentPkg *pkg.Package) bool {
```

Comment:
This function looks intricate enough to warrant some unit tests

Reply:
(And, it's awesome that this was broken out!! 👍 👍 )

Reply:
I thought about it but decided not to for now

> The code was further decomposed as well (the tests were updated, but not decomposed further).

There are tests that exercise this function.

Reply:
Okay, I think I see where you're coming from. The only code path I saw that didn't appear to be tested by the consuming function's test was the `if !ok` path, which is gone now, so that's a moot concern. It looks like this function is getting tested in an indirect way via existing tests, so that's good. My preference would be that we add localized tests for this function in the future, but I'm up for deferring that beyond this PR. Thanks for the consideration.

Reply:
agreed on adding localized tests in the future 👍